worksheet_B/Mandelbrot.cpp

```diff
@@ -17,22 +136,42 @@ int main()
 	// Generate the image
 	for (int pixelY = 0; pixelY < image.height(); pixelY++)
 	{
-		// TODO: Map the y coordinate into the range minY to maxY
-		//double y0 =
+		//Map the y coordinate into the range minY to maxY
+		double y0 = (pixelY / 800.0) * (maxY - minY) + minY;
 
-		for (int pixelX = 0; pixelX < image.width(); pixelX++)
+		for (int pixelX = 0.0; pixelX < image.width(); pixelX++)
 		{
-			// TODO: Map the x coordinate into the range minX to maxX
-			//double x0 =
+			//Map the x coordinate into the range minX to maxX
+			double x0 = (pixelX / 800.0) * (maxX - minX) + minX;
+
+			//algorithm to colour a single pixel (x0, y0) of the Mandelbrot set fractal
+			double yi = 0.0;
+			double xi = 0.0;
+			double loops = 0.0;
+			
+			while (loops < 200 && xi*xi + yi*yi < 4.0)
+			{
+				double tempX = xi*xi - yi*yi + x0;
+				yi = (2 * xi*yi) + y0;
+				xi = tempX;
+				++loops;
 
-			// TODO: implement the algorithm to colour a single pixel (x0, y0) of the Mandelbrot set fractal
-			// The code below simply fills the screen with random pixels
+			}
+			rgb rgbValue;
 
-			// Write the pixel
-			// TODO: change the right-hand side of these three lines to write the desired pixel colour value
-			image(pixelX, pixelY, 0, 0) = rand(); // red component
-			image(pixelX, pixelY, 0, 1) = rand(); // green component
-			image(pixelX, pixelY, 0, 2) = rand(); // blue component
+			if (loops >= 200)                                       
+			{
+				rgbValue = { 0.0, 0.0, 0.0 };                    
+			}
+			else
+			{
+				hsv currentHSV = { loops, 1.0, 1.0 };               
+				rgbValue = hsv2rgb(currentHSV);
+			}
+				//change the right-hand side of these three lines to write the desired pixel colour value
+			image(pixelX, pixelY, 0, 0) = rgbValue.g*255.0 ; // red component
```

Comment:
Worth adding a comment here pointing out that you are swizzling the RGB components for a more pleasing aesthetic effect -- otherwise confusing why you're assigning the "wrong" components